bin/video2x.py

```diff
@@ -298,37 +258,41 @@ def absolutify_paths(config):
 
 # read configurations from JSON
 config = read_config(args.config)
-config = absolutify_paths(config)
 
 # load waifu2x configuration
-if args.driver == 'waifu2x_caffe':
-    waifu2x_settings = config['waifu2x_caffe']
-    if not pathlib.Path(waifu2x_settings['waifu2x_caffe_path']).is_file():
-        Avalon.error('Specified waifu2x-caffe directory doesn\'t exist')
-        Avalon.error('Please check the configuration file settings')
-        raise FileNotFoundError(waifu2x_settings['waifu2x_caffe_path'])
-elif args.driver == 'waifu2x_converter':
-    waifu2x_settings = config['waifu2x_converter']
-    if not pathlib.Path(waifu2x_settings['waifu2x_converter_path']).is_dir():
-        Avalon.error('Specified waifu2x-converter-cpp directory doesn\'t exist')
-        Avalon.error('Please check the configuration file settings')
-        raise FileNotFoundError(waifu2x_settings['waifu2x_converter_path'])
-elif args.driver == 'waifu2x_ncnn_vulkan':
-    waifu2x_settings = config['waifu2x_ncnn_vulkan']
-    if not pathlib.Path(waifu2x_settings['waifu2x_ncnn_vulkan_path']).is_file():
-        Avalon.error('Specified waifu2x_ncnn_vulkan directory doesn\'t exist')
-        Avalon.error('Please check the configuration file settings')
-        raise FileNotFoundError(waifu2x_settings['waifu2x_ncnn_vulkan_path'])
-elif args.driver == 'anime4k':
-    waifu2x_settings = config['anime4k']
-    if not pathlib.Path(waifu2x_settings['anime4k_path']).is_file():
-        Avalon.error('Specified anime4k directory doesn\'t exist')
-        Avalon.error('Please check the configuration file settings')
-        raise FileNotFoundError(waifu2x_settings['anime4k_path'])
+waifu2x_settings = config[args.driver]
+
+# Search for valid java binary path
+if args.driver == 'anime4k':
+    path = common.find_path(waifu2x_settings['java_path'], 'java')
```

Comment:
No, because there might be a specified path in video2x.json, where which will not find.

In common.py I have "command -v" which works like shutil.which.  It may be ideal to switch command -v with shutil.which there.

Reply:
![image](https://user-images.githubusercontent.com/21986859/65475104-d520aa00-de6d-11e9-9923-58a3d75e0849.png)

You can specify a path for shutil.which to search in though.

Reply:
find_path searches multiple paths

Reply:
Overall, there are still a number of things that can be improved in this file.
e.g.
```python
```